File: site/application/models/abstract/Npc.php
<?php
/**
 * Абстрактный класс NPC
 * @author ismd
 */

abstract class Abstract_Npc extends PsModel {

    private $_id;    // id в таблице NpcLayout (конкретный NPC на карте)
    private $_idNpc; // id в таблице Npc (класс NPC)
    private $_name;
    private $_greeting;
    private $_image;
    private $_idLayout;
    private $_x;
    private $_y;

    public function toArray() {
        return array(
            'id'       => $this->getId(),
            'idNpc'    => $this->getIdNpc(),
            'name'     => $this->getName(),
            'greeting' => $this->getGreeting(),
            'image'    => $this->getImage(),
            'idLayout' => $this->getIdLayout(),
            'x'        => $this->getX(),
            'y'        => $this->getY(),
        );
    }

    public function setId($value) {
        $this->_id = (int)$value;
        return $this;
    }

    public function getId() {
        return $this->_id;
    }

    public function setIdNpc($value) {
        $this->_idNpc = (int)$value;
        return $this;
    }

    public function getIdNpc() {
        return $this->_idNpc;
    }

    public function setName($value) {
        $this->_name = (string)$value;
        return $this;
    }

    public function getName() {
        return $this->_name;
    }

    public function setGreeting($value) {
        $this->_greeting = (string)$value;
        return $this;
    }

    public function getGreeting() {
        return $this->_greeting;
    }

    public function setImage($value) {
        $this->_image = (string)$value;
        return $this;
    }

    public function getImage() {
        return $this->_image;
    }

    public function setIdLayout($value) {
        $this->_idLayout = (int)$value;
        return $this;
    }

    public function getIdLayout() {
        return $this->_idLayout;
    }

    public function setX($value) {
        $this->_x = (int)$value;
        return $this;
    }

    public function getX() {
        return $this->_x;
    }

    public function setY($value) {
        $this->_y = (int)$value;
        return $this;
    }

    public function getY() {
        return $this->_y;
    }
}

File: game/src/main/java/game/npc/Npc.java
package game.npc;

import game.layout.CellContent;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;

/**
 * @author ismd
 */
@Entity
public class Npc extends CellContent {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private int id;
    private String name;
    private String greeting;
    private String image;
    private int idLayout;
    private int x;
    private int y;

    public Npc() {
    }

    public int getId() {
        return id;
    }

    public Npc setId(int id) {
        this.id = id;
        return this;
    }

    public String getName() {
        return name;
    }

    public Npc setName(String name) {
        this.name = name;
        return this;
    }

    public String getGreeting() {
        return greeting;
    }

    public Npc setGreeting(String greeting) {
        this.greeting = greeting;
        return this;
    }

    public String getImage() {
        return image;
    }

    public Npc setImage(String image) {
        this.image = image;
        return this;
    }

    public int getIdLayout() {
        return idLayout;
    }

    public Npc setIdLayout(int idLayout) {
        this.idLayout = idLayout;
        return this;
    }

    public int getX() {
        return x;
    }

    public Npc setX(int x) {
        this.x = x;
        return this;
    }

    public int getY() {
        return y;
    }

    public Npc setY(int y) {
        this.y = y;
        return this;
    }
}
